File: SampleApp/Objective-C/SampleApp/Banner/BannerNavigationController.m
//
//  BannerNavigationController.m
//  SampleApp
//
//  Fluct SDK
//  Copyright © 2016年 fluct, Inc. All rights reserved.
//

#import "BannerNavigationController.h"
#import <FluctSDK/FluctSDK.h>

@interface BannerNavigationController ()
@property (nonatomic) FSSAdView *adView;
@end

@implementation BannerNavigationController

- (void)viewDidLoad {
    [super viewDidLoad];
    FSSAdView *adView = [[FSSAdView alloc] initWithGroupId:@"1000055927" unitId:@"1000084701" adSize:FSSAdSize320x50];
    [self.view addSubview:adView];
    [adView loadAd];
    self.adView = adView;
}

- (void)viewDidLayoutSubviews {
    [super viewDidLayoutSubviews];
    CGFloat adViewHeight = CGRectGetHeight(self.adView.frame);
    CGFloat maxY = CGRectGetMaxY(self.view.bounds);
    CGFloat adViewY = maxY - self.view.layoutMargins.bottom - adViewHeight;

    CGFloat midX = CGRectGetMidX(self.view.bounds);
    CGFloat adViewX = midX - CGRectGetWidth(self.adView.frame) * 0.5;

    CGRect adViewFrame = self.adView.frame;
    adViewFrame.origin = CGPointMake(adViewX, adViewY);
    self.adView.frame = adViewFrame;
}

@end

File: SampleApp/Swift/SDK/SampleApp/Banner/BannerNavigationController.swift
//
//  BannerNavigationController.swift
//  SampleApp
//
//  Fluct SDK
//  Copyright © 2016年 fluct, Inc. All rights reserved.
//

import UIKit
import FluctSDK

class BannerNavigationController: UINavigationController {

    private var adView: FSSAdView?

    override func viewDidLoad() {
        super.viewDidLoad()
        let adView = FSSAdView(groupId: "1000055927", unitId: "1000084701", adSize: FSSAdSize320x50)
        self.view.addSubview(adView)
        adView.loadAd()
        self.adView = adView
    }

    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()

        let adViewHeight = self.adView?.frame.height ?? 0.0
        let maxY = self.view.bounds.maxY
        let adViewY = maxY - self.view.layoutMargins.bottom - adViewHeight

        let adViewWidth = self.adView?.frame.width ?? 0.0
        let midX = self.view.bounds.midX
        let adViewX = midX - adViewWidth * 0.5

        var frame = adView?.frame ?? .zero
        frame.origin = CGPoint(x: adViewX, y: adViewY)
        adView?.frame = frame
    }

}
